docs/core/testing/unit-testing-best-practices.md

```diff
@@ -0,0 +1,515 @@
+{
+    var glossary = new Glossary();
+
+    Assert.False(glossary.IsValidWord(null)); // potential NullException
+    Assert.False(glossary.IsValidWord("")); // this never gets executed if the above test fails
+}
+```
+
+#### Better:
+```csharp
+[InlineData(null)]
+[InlineData("")]
+public void IsValidWord_InputIsNullOrEmpty_ReturnsFalse(string input)
+{
+    var glossary = new Glossary();
+    
+    var result = glossary.IsValidWord(input);
+
+    Assert.False(result);
+}
+```
+
+### Write Minimally Passing Tests
+The input to be used in a unit test should be the simplest possible in order to pass the behavior that you are currently testing.
+
+#### Why?
+- Tests become more resilient to future changes in the codebase.
+- Closer to testing behavior over implementation.
+
+#### Bad:
+```csharp
+public void ConcatenateWords_ByDefault_ReturnsStringWithCommaBetween()
+{
+    var glossary = new Glossary();
+    var firstWord = "aardvark";
+    var secondWord = "baboon";
+
+    var result = glossary.ConcatenateWords(firstWord, secondWord)
+
+    Assert.Equals("aardvark,baboon", result)
+}
+```
+
+#### Better:
+```csharp
+public void ConcatenateWords_ByDefault_ReturnsStringWithCommaBetween()
+{
+    var glossary = new Glossary();
+
+    var result = glossary.ConcatenateWords("a", "b")
+
+    Assert.Equals("a,b", result)
+}
+```
+
+## How Do I...?
+
+### Test Private Methods
+In most cases, there should not be a need to test a private method. Private methods are an implementation detail. You can think of it this way: private methods never exist in isolation. At some point, there is going to be a public facing method that calls the private method as part of its implementation. What you should care about is the end result of the public method that calls into the private one. 
+
+Consider the following case
+
+```csharp
+public string ParseLogLine(string input)
+{
+    var sanitizedInput = trimInput(input);
+    return sanitizedInput;
+}
+
+private string trimInput(string input)
+{
+    return input.Trim();
+}
+```
+
+Your first reaction may be to start writing a test for `trimInput` because you want to make sure that the method is working as expected. However, it is entirely possible that `ParseLogLine` manipulates `sanitizedInput` in such a way that we do not expect, rendering a test against `trimInput` useless. 
+
+The real test should be done against the public facing method `ParseLogLine` because that is what we ultimately care about. 
+
+```csharp
+public void ParseLogLine_ByDefault_ReturnsTrimmedResult()
+{
+    var parser = new Parser();
+
+    var result = parser.ParseLogLine(" a ");
+
+    Assert.Equals("a", result);
+}
+```
+
+With this viewpoint, if you see a private method, find the public method and write your tests against that method. Just because a private method returns the expected result, does not mean the system that eventually calls the private method uses the result correctly.
+
+### Stub Static References
```

Comment:
I like this example, but I'd like to see you set it up more by using a realistic scenario: Maybe something is half price on Tuesday.